object_manager.go

```diff
@@ -301,3 +556,93 @@ func (objMgr *ObjectManager) UpdateZoneDelegated(ref string, delegate_to []NameS
 func (objMgr *ObjectManager) DeleteZoneDelegated(ref string) (string, error) {
 	return objMgr.connector.DeleteObject(ref)
 }
+
+// SearchObjectByAltId is a generic function to search object by alternate id
+func (objMgr *ObjectManager) SearchObjectByAltId(
+	objType string, ref string, internalId string, eaNameForInternalId string) (interface{}, error) {
+	var (
+		err        error
+		recordType IBObject
+		res        interface{}
+	)
+	val, ok := getRecordTypeMap[objType]
+	if !ok {
+		return nil, fmt.Errorf("unknown record type")
+	}
+	recordType = val(ref)
+
+	if ref != "" {
+		// Fetching object by reference
+		if err := objMgr.connector.GetObject(recordType, ref, NewQueryParams(false, nil), &res); err != nil {
+			if _, ok := err.(*NotFoundError); !ok {
+				return nil, err
+			}
+		}
+		success, err := validateObjByInternalId(res, internalId, eaNameForInternalId)
+		if err != nil {
+			return nil, err
+		} else if success {
+			return res, nil
+		}
+	}
+
+	sf := map[string]string{
+		fmt.Sprintf("*%s", eaNameForInternalId): internalId,
+	}
+
+	// Fetch the object by search fields
+	getObjectWithSearchFields, ok := getObjectWithSearchFieldsMap[objType]
+	if !ok {
+		return nil, fmt.Errorf("unknown record type")
+	}
+	res, err = getObjectWithSearchFields(recordType, objMgr, sf)
+	if err != nil {
+		return nil, err
+	}
+	if res == nil {
+		return nil, NewNotFoundError("record not found")
+	}
+
+	return &res, nil
+}
+
+// validateObjByInternalId validates the object by comparing the given internal with the object's internal id
+func validateObjByInternalId(res interface{}, internalId, eaNameForInternalId string) (bool, error) {
+	var success bool
+	if res == nil {
+		return success, nil
+	} else if strings.TrimSpace(internalId) == "" {
+		// return object if internal id is empty
+		success = true
+		return success, nil
```

Comment:
why it is returned true here

Reply:
We should return the return response received when fetched by ref and the internal id is empty. I should have updated the comment to `return true if response is not nil and internal id is empty`.